### DISCUSSION TITLE
ESP32 OpenThread API

### DISCUSSION THREAD

**Discussion by im22361** [openthread/openthread#10231] [Q&A] [answered]
Hello, 

I am trying to use the "Developing with OpenThread APIs guide"(https://openthread.io/codelabs/openthread-apis#2) to develop an OPI that will send sensor data.

Although the example is for Nordic this is the only example I can see. Is there an example for ESP32? 

I have tried to use the code found in the guide, but when I do I get an error with #include "utils/code_utils.h". Is there a way to fix this with EP32?

**Comment by chshu** [comment]
@im22361 The OpenThread APIs are common to all platforms.

Could you follow this instruction to setup ESP platform: https://openthread.io/codelabs/esp-openthread-hardware. Feel free to raise issues here: https://github.com/espressif/esp-idf/issues for any development questions.

**Reply by im22361** [reply]
Hi, 
Thank you for the response. I have already set up the ESP platform and I can send messages between the ESP32 devices. 

I am following OpenThread guide: https://openthread.io/codelabs/openthread-apis#2
It says to include utils/code_utils.h
Is this an OpenThread specific library?

**Reply by jwhui** [reply, marked as answer]
https://github.com/openthread/openthread/blob/main/examples/platforms/utils/code_utils.h
